pkg/basicstation/cups/update_info.go

```diff
@@ -188,12 +192,14 @@ func (s *Server) UpdateInfo(c echo.Context) error {
 			// Use the generated CUPS API Key for authenticating subsequent calls.
 			md := metadata.New(map[string]string{
 				"id":            ids.GatewayID,
-				"authorization": fmt.Sprintf("Bearer %s", gtw.Attributes[cupsCredentialsAttribute]),
+				"authorization": fmt.Sprintf("Bearer %s", string(gtw.TargetCUPSKey.Value)),
 			})
 			if ctxMd, ok := metadata.FromIncomingContext(ctx); ok {
 				md = metadata.Join(ctxMd, md)
 			}
 			ctx = metadata.NewIncomingContext(ctx, md)
+			// This ensures that the newly registered gateway will make a CUPS request to the `gtw.TargetCUPSURI`.
```

Comment:
I would reword this a bit, as we don't know for sure what the gateway will do. We do know that this will cause the server to return the CUPS URI.